### ISSUE TITLE
error start msf_visualizer

### ISSUE THREAD

**Issue by AlexandrZabolotny** [ApolloAuto/apollo#12735]
Hi team!
My OS - Ubuntu 18.04.
I use apollo-6.0

I am trying to visualize msf_localization accoding [this](https://github.com/ApolloAuto/apollo/blob/master/docs/howto/how_to_run_MSF_localization_module_on_your_local_computer.md)

Then I run **mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag**  
I get error

WARNING: Logging before InitGoogleLogging() is written to STDERR
I1006 14:04:22.833047 11027 module_argument.cc:81] []command: mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag 
I1006 14:04:22.833330 11027 global_data.cc:153] []host ip: 10.152.1.8
I1006 14:04:22.833494 11027 module_argument.cc:57] []binary_name_ is mainboard, process_group_ is mainboard_default, has 1 dag conf
I1006 14:04:22.833501 11027 module_argument.cc:60] []dag_conf: /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag
Load lidar extrinsic succeed.
Load map config succeed.
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.4.0) /tmp/installers/opencv-4.4.0/modules/highgui/src/window.cpp:634: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvNamedWindow'

Aborted (core dumped)

**Comment by storypku** [comment]
> Hi team!
> My OS - Ubuntu 18.04.
> I use apollo-6.0
> 
> I am trying to visualize msf_localization accoding [this](https://github.com/ApolloAuto/apollo/blob/master/docs/howto/how_to_run_MSF_localization_module_on_your_local_computer.md)
> 
> Then I run **mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag**
> I get error
> 
> WARNING: Logging before InitGoogleLogging() is written to STDERR
> I1006 14:04:22.833047 11027 module_argument.cc:81] []command: mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag
> I1006 14:04:22.833330 11027 global_data.cc:153] []host ip: 10.152.1.8
> I1006 14:04:22.833494 11027 module_argument.cc:57] []binary_name_ is mainboard, process_group_ is mainboard_default, has 1 dag conf
> I1006 14:04:22.833501 11027 module_argument.cc:60] []dag_conf: /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag
> Load lidar extrinsic succeed.
> Load map config succeed.
> terminate called after throwing an instance of 'cv::Exception'
> what(): OpenCV(4.4.0) /tmp/installers/opencv-4.4.0/modules/highgui/src/window.cpp:634: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvNamedWindow'
> 
> Aborted (core dumped)

It seems that OpenCV was built without GTK+ 2.x support. You can rebuild OpenCV with gtk2.0 support. The team will fix it ASAP. Thanks for reporting.

**Comment by AlexandrZabolotny** [comment]
Can I use [this](https://github.com/ApolloAuto/apollo/blob/master/docker/build/installers/install_opencv.sh) script to solve this problem?

**Comment by storypku** [comment]
Yep,  just append `libgtk2.0-dev` at the apt_get_update_and_install section.

**Comment by AlexandrZabolotny** [comment]
> Yep, just append `libgtk2.0-dev` at the apt_get_update_and_install section.

1. I append `libgtk2.0-dev` at the apt_get_update_and_install section
2. run in docker append sudo bash /apollo/docker/build/installers/install_opencv.sh
3. [ OK ] Successfully installed OpenCV 4.4.0

But the error persisted

**Comment by storypku** [comment]
I personally was on vocation till next Monday. You can try change `-DWITH_GTK=OFF` to `-DWITH_GTK=ON` and try again.

**Comment by AlexandrZabolotny** [comment]
> I personally was on vocation till next Monday. You can try change `-DWITH_GTK=OFF` to `-DWITH_GTK=ON` and
> set `-DWITH_GTK_2_X=ON` also, and try agin.

I followed your recommendations, but the problem was not resolved.

Perhaps you can tell me how to display the local_map (obtained using this [script](https://github.com/ApolloAuto/apollo/blob/master/scripts/msf_simple_map_creator.sh)). This will partially solve my problem. I want to see the local_map as shown in the picture below. 

![Screenshot from 2020-09-25 10-04-06](https://user-images.githubusercontent.com/61351599/95730898-8a93ea80-0c87-11eb-910e-fb7a65bdb91b.png)

**Comment by storypku** [comment]
@guoweiwan , can you help with this? I personally can't executing 

```
mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag
```
also.

**Comment by devynliu** [comment]
hello，l recently meet this issue. I have fix it. i append libgtk2.0-dev , libqt4-dev ;  -DWITH_GTK=OFF to -DWITH_GTK=ON and
set -DWITH_GTK_2_X=ON set -DWITH_QT=ON. then re_run .

**Comment by AlexandrZabolotny** [comment]
> hello，l recently meet this issue. I have fix it. i append libgtk2.0-dev , libqt4-dev ; -DWITH_GTK=OFF to -DWITH_GTK=ON and
> set -DWITH_GTK_2_X=ON set -DWITH_QT=ON. then re_run .

Hello @devynliu   
Thank You very much! This is the solution to given problem.  
But now I must to install OpenCV after every reboot of my PC. Probably because I have no experience with docker.
Can you tell me how to save compiled libraries in Apollo docker?

**Comment by AlexandrZabolotny** [comment]
**Hello colleagues!  
I was able to install the library.  

But now another error has appeared. Unfortunately, information about it is not logged. Maybe someone has some ideas?**


@in-dev-docker:/apollo]$ mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1113 11:05:16.417951  2781 module_argument.cc:81] []command: mainboard -d /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag 
I1113 11:05:16.419196  2781 global_data.cc:153] []host ip: 10.152.1.8
I1113 11:05:16.419415  2781 module_argument.cc:57] []binary_name_ is mainboard, process_group_ is mainboard_default, has 1 dag conf
I1113 11:05:16.419425  2781 module_argument.cc:60] []dag_conf: /apollo/modules/localization/dag/dag_streaming_msf_visualizer.dag
Load lidar extrinsic succeed.
Load map config succeed.
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-shatl'
E1113 11:05:18.004359  2790 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.007820  2790 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.083961  2792 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.084118  2792 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.161653  2797 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.161803  2797 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.256880  2783 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.257038  2783 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.355005  2786 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.355154  2786 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.453400  2790 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.453462  2790 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.462996  2811 visualization_manager.cc:241] [localization_visualizer]The queried buffer is empty.
E1113 11:05:18.540704  2796 online_visualizer_component.cc:144] [mainboard]start
E1113 11:05:18.540843  2796 online_visualizer_component.cc:146] [mainboard]stop
E1113 11:05:18.543491  2811 visualization_manager.cc:241] [localization_visualizer]The queried buffer is empty.
E1113 11:05:18.543517  2811 visualization_manager.cc:95] [localization_visualizer]The buffer is empty.
Segmentation fault (core dumped)

**Comment by mengen100** [comment]
> localization_visualizer

@AlexandrZabolotny Hi Alex, I'm facing the same issue when I first start the visualizer which is black screen interface (not sure if that's the correct behavior?) and then followed by starting the LGSVL simulator simulation, the command prompt then pops a lot of "The queried buffer is empty." errors. 
Have you found a solution towards this issue?

**Comment by storypku** [comment]
> > hello，l recently meet this issue. I have fix it. i append libgtk2.0-dev , libqt4-dev ; -DWITH_GTK=OFF to -DWITH_GTK=ON and
> > set -DWITH_GTK_2_X=ON set -DWITH_QT=ON. then re_run .
> 
> Hello @devynliu
> Thank You very much! This is the solution to given problem.
> But now I must to install OpenCV after every reboot of my PC. Probably because I have no experience with docker.
> Can you tell me how to save compiled libraries in Apollo docker?

Hi @AlexandrZabolotny, the build issue was resolved in PR #13730 and I am going to release a new Docker image for the properly built OpenCV.
